### ISSUE TITLE
Ambients are not transferred across pop ups

### ISSUE THREAD

**Issue by Dominaezzz** [JetBrains/compose-jb#135]
This is a reproducer.
```kotlin
fun main() {
    val testAmbient = staticAmbientOf<String> { error("Shouldn't be called.") }
    Window(title = "My ambient popup test") {
        Providers(testAmbient provides "TESTING") {
            Popup {
                // Exception in thread "AWT-EventQueue-0" java.lang.IllegalStateException: Shouldn't be called.
                Text(testAmbient.current)
            }
        }
    }
}
```

This throws `Exception in thread "AWT-EventQueue-0" java.lang.IllegalStateException: Shouldn't be called.`

Is this expected behaviour?

**Comment by olonho** [comment]
Does the same code works in Android as you expected?

**Comment by Dominaezzz** [comment]
I haven't used compose on Android, so I don't know if it works any differently there.
Just based on the documentation alone, this looks like a bug. Either code or docs need to be fixed.

**Comment by igordmn** [comment]
I checked, on Android it works

**Comment by igordmn** [comment]
Fixed in 0.3.0-build135

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5598">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
